src/DIRAC/MonitoringSystem/Client/DataOperationSender.py

```diff
@@ -41,16 +41,27 @@ def __init__(self):
             self._sendDataMethods.append(getattr(self, f"_sendData{backend}"))
             self._commitMethods.append(getattr(self, f"_commit{backend}"))
 
-    def _sendDataMonitoring(self, baseDict, commitFlag=False, delayedCommit=False, startTime=False, endTime=False):
+    def _sendDataMonitoring(
+        self, baseDict, commitFlag=False, delayedCommit=False, startTime=False, endTime=False, failedRecords=None
+    ):
         """Send the data to the monitoring system"""
-        # Some fields added here are not known to the Accounting, so we have to make a copy
-        monitoringDict = copy.deepcopy(baseDict)
 
-        monitoringDict["Channel"] = monitoringDict["Source"] + "->" + monitoringDict["Destination"]
+        # Since we are adding elements that the accounting
+        # may not like, work on a copy
+        baseDict = copy.copy(baseDict)
+        baseDict["Channel"] = baseDict["Source"] + "->" + baseDict["Destination"]
         # Add timestamp if not already added
-        if "timestamp" not in monitoringDict:
-            monitoringDict["timestamp"] = int(toEpochMilliSeconds())
-        self.dataOperationReporter.addRecord(monitoringDict)
+        if "timestamp" not in baseDict:
+            baseDict["timestamp"] = int(toEpochMilliSeconds())
+        self.dataOperationReporter.addRecord(baseDict)
+
+        # If there were failedRecords, send them right away
+        if failedRecords:
+            for failedRec in failedRecords:
+                self.failedDataOperationReporter.addRecord(failedRec)
+
+            self.failedDataOperationReporter.commit()
```

Comment:
There is a `commitFlag` in this method which is used a few lines below. I don't see right now a need for NOT committing straight away, but I am wondering if you have evaluated about it.

Reply:
I guess this was added for consistency with what was in the accounting. In principle, I would be fine to NOT have the commit flag for the monitoring, but then there's the risk of having data in the monitoring which is not in the accounting, which is something we do not want if the accounting is first in the list.
I am happy to remove that flag for 8.1

Reply:
I think it's also fine as it is now.